apps/src/aichat/views/ChatWorkspace.tsx

```diff
@@ -45,42 +45,31 @@ const ChatWorkspace: React.FunctionComponent<ChatWorkspaceProps> = ({
   const [selectedTab, setSelectedTab] =
     useState<WorkspaceTeacherViewTab | null>(null);
 
-  const {showWarningModal, isWaitingForChatResponse} = useAppSelector(
-    state => state.aichat
-  );
+  const {showWarningModal, isWaitingForChatResponse, studentChatHistory} =
+    useAppSelector(state => state.aichat);
   const viewAsUserId = useAppSelector(state => state.progress.viewAsUserId);
-  const items = useSelector(selectAllMessages);
+  const currentLevelId = useAppSelector(state => state.progress.currentLevelId);
+  const visibleItems = useSelector(selectAllVisibleMessages);
 
   const students = useSelector(
     (state: {teacherSections: {selectedStudents: Students}}) =>
       state.teacherSections.selectedStudents
   );
 
-  // Compare the messages as a string since the object reference will change on every update.
-  // This way we will only scroll when the contents of the messages have changed.
-  const messagesString = JSON.stringify(items);
-  const conversationContainerRef = useRef<HTMLDivElement>(null);
-
-  useEffect(() => {
-    if (conversationContainerRef.current) {
-      conversationContainerRef.current.scrollTo({
-        top: conversationContainerRef.current.scrollHeight,
-        behavior: 'smooth',
-      });
-    }
-  }, [messagesString, isWaitingForChatResponse]);
+  const dispatch = useAppDispatch();
 
   const selectedStudentName = useMemo(() => {
-    if (viewAsUserId) {
+    if (viewAsUserId && currentLevelId) {
       const selectedStudent = Object.values(students).find(
         student => student.id === viewAsUserId
       );
       if (selectedStudent) {
+        dispatch(fetchStudentChatHistory(selectedStudent.id));
         return getShortName(selectedStudent.name);
       }
     }
     return null;
-  }, [viewAsUserId, students]);
+  }, [viewAsUserId, students, dispatch, currentLevelId]);
```

Comment:
FYI - added `currentLevelId` so that student chat history is updated when level changes (after pr approval) cc: @sanchitmalhotra126